builtin/providers/aws/validators.go

```diff
@@ -1154,3 +1154,19 @@ func validateDbOptionGroupNamePrefix(v interface{}, k string) (ws []string, erro
 	}
 	return
 }
+
+func validateAwsAlbTargetGroupName(v interface{}, k string) (ws []string, errors []error) {
+	name := v.(string)
+	if len(name) > 32 {
+		errors = append(errors, fmt.Errorf("%q (%q) cannot be longer than '32' characters", k, name))
+	}
+	return
+}
+
+func validateAwsAlbTargetGroupNamePrefix(v interface{}, k string) (ws []string, errors []error) {
+	name := v.(string)
+	if len(name) > 32 {
```

Comment:
See the discussion https://github.com/hashicorp/terraform/pull/13442/files#r110410247